modules/charon-core/src/test/java/org/wso2/charon3/core/schema/ServerSideValidatorTest.java

```diff
@@ -0,0 +1,663 @@
+    public Object[][] dataToValidateRetrievedSCIMObjectInList() throws InstantiationException, IllegalAccessException {
+
+        return new Object[][]{
+
+                {createNewUser(), createSCIMResourceTypeSchemaUser(), "name", "emails"}
+        };
+    }
+
+    @Test(dataProvider = "dataForValidateRetrievedSCIMObjectInList")
+    public void testValidateRetrievedSCIMObjectInList(Object objectScimObject, Object objectResourceSchema,
+            Object objectRequestedAttributes, Object objectRequestedExcludingAttributes)
+            throws BadRequestException, CharonException {
+
+        User scimObject = (User) objectScimObject;
+        SCIMResourceTypeSchema resourceSchema = (SCIMResourceTypeSchema) objectResourceSchema;
+        String requestedAttributes = (String) objectRequestedAttributes;
+        String requestedExcludingAttributes = (String) objectRequestedExcludingAttributes;
+
+        ServerSideValidator.validateRetrievedSCIMObjectInList(scimObject, resourceSchema, requestedAttributes,
+                requestedExcludingAttributes);
+
+        Assert.assertTrue(true, "validateRetrievedSCIMObjectInList is successful");
+    }
+
+    @DataProvider(name = "dataForValidateRetrievedSCIMObject")
+    public Object[][] dataToValidateRetrievedSCIMObject() throws InstantiationException, IllegalAccessException {
+
+        return new Object[][]{
+
+                {createNewUser(), createSCIMResourceTypeSchemaUser(), "name", "emails"}
+        };
+    }
+
+    @Test(dataProvider = "dataForValidateRetrievedSCIMObject")
+    public void testValidateRetrievedSCIMObject(Object objectScimObject, Object objectResourceSchema,
+            Object objectRequestedAttributes, Object objectRequestedExcludingAttributes)
+            throws BadRequestException, CharonException {
+
+        User scimObject = (User) objectScimObject;
+        SCIMResourceTypeSchema resourceSchema = (SCIMResourceTypeSchema) objectResourceSchema;
+        String requestedAttributes = (String) objectRequestedAttributes;
+        String requestedExcludingAttributes = (String) objectRequestedExcludingAttributes;
+
+        ServerSideValidator.validateRetrievedSCIMObject(scimObject, resourceSchema, requestedAttributes,
+                requestedExcludingAttributes);
+
+        Assert.assertTrue(true, "validateRetrievedSCIMObject is successful");
+    }
+
+    @DataProvider(name = "dataForValidateRetrievedSCIMRoleObject")
+    public Object[][] dataToValidateRetrievedSCIMRoleObject() throws InstantiationException, IllegalAccessException {
+
+        Role role = createNewRole();
+
+        return new Object[][]{
+
+                {role, "displayName", "groups"},
+                {role, null, "users"},
+                {role, null, null}
+        };
+    }
+
+    @Test(dataProvider = "dataForValidateRetrievedSCIMRoleObject")
+    public void testValidateRetrievedSCIMRoleObject(Object objectScimObject, Object objectRequestedAttributes,
```

Comment:
Some of these parameters seems to be strings. Can't we define them as strings, rather than objects?

Check other places as well.

Reply:
fixed in abc18d12f649b0a2709ddcde8561827e72d4cc36